### PR TITLE
Correct understanding of ask timeouts, remove dead code.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Comment by scottfrazer** [comment]
:+1:

File: src/main/scala/cromwell/Main.scala [hunk 1]
```diff
@@ -42,7 +42,11 @@ object Main extends App {
    * Also now passing args to runAction instead of the constructor, as even sbt seemed to have issues with the args
    * array becoming null in "new Main(args)" when used with: sbt 'run run ...'
    */
-  val AskTimeout = Timeout(5 seconds)
+  // The magic maximum value for an ask timeout.  This value should be large to represent the amount of time until
+  // the future will fail (not the amount of time to wait for the Future to be returned from the ask).  This Future
+  // completes after the workflow goes to a terminal status, so the timeout must be at least as long as the workflow
+  // run duration.
+  val AskTimeout = Timeout(21474835 seconds)
```
Comment (geoffjentry) [inline src/main/scala/cromwell/Main.scala]: Maybe this is a sign that ask isn't the right thing to be doing? This implies you could never have a workflow which takes that long (I realize that having an ~8 month workflow running via command line would be a bit insane)

Comment (mcovarr) [inline src/main/scala/cromwell/Main.scala]: This needs to bridge the non-actor Main to the Cromwell actor system.  The simplest form of synchronization would seem to be blocking on the completion of a `Promise`, which I believe is essentially what `ask`'s `PromiseActorRef` is [giving us](https://github.com/akka/akka/blob/a7021827c3974762143920139a72f33064dd78aa/akka-actor/src/main/scala/akka/pattern/AskSupport.scala#L597).

Granted this seems like a weird usage, but it is a weird use case.  :smile:

Comment (kshakir) [inline src/main/scala/cromwell/Main.scala]: :+1: to @geoffjentry 

Right now timeouts are specified throughout `Main`, in:

- `Main$.<init>`
- `Main.run`
- `Main.runAction`
- `Main.runWorkflow`

One could consolidate the use of timeouts into a single line within `runWorkflow`:

```scala
val runner = workflowManagerSystem.actorSystem.actorOf(runnerProps, "SingleWorkflowRunnerActor")

val futureResult = runner.ask(RunWorkflow)(Timeout(21474835 seconds))
Await.ready(futureResult, Duration.Inf)
```

Alternatively, as was discussed, one could use a small custom `Actor`, just like the `PromiseActorRef`, but that doesn't timeout.

```scala
val runner = workflowManagerSystem.actorSystem.actorOf(runnerProps, "SingleWorkflowRunnerActor")

val promise = Promise[Any]()
workflowManagerSystem.actorSystem.actorOf(Props(classOf[PromiseWorkflowActor], promise, runner))
val futureResult = promise.future
Await.ready(futureResult, Duration.Inf)
```

Where the actor looks something like:

```scala
class PromiseWorkflowActor(promise: Promise[Any], runner: ActorRef) extends Actor {
  runner ! RunWorkflow

  override def receive = {
    case Status.Failure(f) => promise.tryFailure(f)
    case success => promise.trySuccess(success)
  }
}
```

**Comment by kshakir** [comment]
Commented on at least simplifying the timeout code, possibly removing it altogether.

Back to @mcovarr

**Comment by mcovarr** [comment]
Many thanks @kshakir, that's a big improvement.  The wheel chose my reviewer wisely.  :smile:

**Comment by kshakir** [comment]
:+1:

**Pull request merged**